plugins/@grouparoo/pardot/src/lib/export/exportProfiles.ts

```diff
@@ -0,0 +1,273 @@
+
+  // Sometimes we may get "414 - URI Too Long" from Pardot.
+  // In extremely long URIs, 414 won't be returned and "400 - Bad Request" will be returned instead.
+  // In these cases, we recursively split into two batches...
+  try {
+    await updateProspects(client, users, cacheData);
+  } catch (error) {
+    if (error.response?.status === 414 || error.response?.status === 400) {
+      if (users.length === 1) {
+        users[0].error = new Error(
+          `${error.response.status} - ${error.response.statusText}`
+        );
+        return;
+      }
+
+      const half = Math.ceil(users.length / 2);
+      await updateProspectsHelper(client, users.slice(0, half), cacheData);
+      await updateProspectsHelper(client, users.slice(half), cacheData);
+    } else {
+      throw error;
+    }
+  }
+}
+
+async function updateProspects(
+  client: PardotClient,
+  users: BatchExport[],
+  cacheData: PardotCacheData
+) {
+  const input: any[] = [];
+  for (const exportedProfile of users) {
+    const payload = await buildPayload(client, exportedProfile, cacheData);
+    input.push(payload);
+  }
+
+  const errors = await client.batchUpsertProspects(input);
+  for (let errorIdx in errors) {
+    const resultIdx = parseInt(errorIdx);
+    const user = users[resultIdx];
+    user.error = new Error(errors[errorIdx]);
+  }
+}
+
+async function buildPayload(
+  client: PardotClient,
+  exportedProfile: BatchExport,
+  cacheData: PardotCacheData
+) {
+  const {
+    destinationId,
+    oldProfileProperties,
+    newProfileProperties,
+    oldGroups,
+    newGroups,
+    foreignKeyValue,
+  } = exportedProfile;
+
+  const payload: any = {};
+
+  payload.email = foreignKeyValue; // email
+  if (destinationId) {
+    payload.id = destinationId;
+  }
+
+  if (!payload.email) {
+    throw new Error(`newProfileProperties[email] is a required mapping`);
+  }
+
+  // set profile properties, including old ones
+  const newKeys = Object.keys(newProfileProperties);
+  const oldKeys = Object.keys(oldProfileProperties);
+  const allKeys = new Set([...oldKeys, ...newKeys]);
+
+  for (const key of allKeys) {
+    if (["id", "email"].includes(key)) {
+      continue; // set above
+    }
+    const value = newProfileProperties[key]; // includes clearing out removed ones (by setting to null)
+    payload[key] = formatVar(value);
+  }
+
+  // Lists can be set as properties in the payload
+  // list_123=1 will include the user in list ID 123, and list_123=0 will remove them
+
+  for (const group of newGroups) {
+    const listId = await getListId(client, cacheData, group, true);
+    payload[`list_${listId}`] = 1;
+  }
+
+  for (const group of oldGroups) {
+    if (!newGroups.includes(group)) {
+      const listId = await getListId(client, cacheData, group, false);
+      if (listId) {
+        payload[`list_${listId}`] = 0;
+      }
+    }
+  }
+
+  return payload;
+}
+
+function formatVar(value) {
+  if (!value) {
+    return ""; // empty string clears the value
```

Comment:
does null work here?

Reply:
Nope, it just leaves the old value and does nothing. `false` doesn't work either.